In [33]:
import pandas as pd
# pd.set_option('max columns', 20)

# IT Salary Survey EU 2018
## 1. Read CSV file

In [34]:
df_2018 = pd.read_csv("IT Salary Survey EU 2018.csv")

## i. Data Understanding
* **Shape** ```df.shape``` - tells us the size of our data frame. Row * Column (Like the order of a matrix)
* **Head & Tail** ```df.head()```, ```df.tail()``` - shows us the first/last 5 rows of data
* **Columns** ```df.columns()``` - used to list all the columns

In [35]:
print(df_2018.shape)

(765, 14)


Print the first and last 5 lines of the csv data to know the size of the data we're working with

In [36]:
# First 5
print(df_2018.head())

             Timestamp   Age Gender     City                       Position  \
0  14/12/2018 12:41:33  43.0      M  München                   QA Ingenieur   
1  14/12/2018 12:42:09  33.0      F  München  Senior PHP Magento developer    
2  14/12/2018 12:47:36  32.0      M  München              Software Engineer   
3  14/12/2018 12:50:15  25.0      M  München      Senior Frontend Developer   
4  14/12/2018 12:50:31  39.0      M  München                    UX Designer   

   Years of experience Your level  Current Salary  Salary one year ago  \
0                 11.0     Senior         77000.0              76200.0   
1                  8.0     Senior         65000.0              55000.0   
2                 10.0     Senior         88000.0              73000.0   
3                  6.0     Senior         78000.0              55000.0   
4                 10.0     Senior         69000.0              60000.0   

   Salary two years ago Are you getting any Stock Options?  \
0               68

In [37]:
# Last 5
print(df_2018.tail())

               Timestamp   Age Gender     City               Position  \
760  03/06/2020 20:12:51  40.0      M     Köln  Java Developer junior   
761  28/07/2020 04:03:13   NaN      M     Köln      E.g. C# Developer   
762  28/07/2020 04:03:20   NaN      M     Köln      E.g. C# Developer   
763  26/08/2020 09:06:44   NaN    NaN      NaN                    NaN   
764  21/09/2020 01:47:10  31.0      F  München                    Pm    

     Years of experience Your level  Current Salary  Salary one year ago  \
760                  1.0     Junior         44000.0              40000.0   
761                  1.0     Junior         45000.0              40000.0   
762                  1.0     Junior         45000.0              40000.0   
763                  NaN        NaN             NaN                  NaN   
764                 10.0     Senior        110000.0                  NaN   

     Salary two years ago Are you getting any Stock Options?  \
760               40000.0               

Since our first entry is at index 0, we have 764 entries in total.